standard-controller/src/main/java/io/enmasse/controller/standard/AddressController.java

```diff
@@ -294,6 +316,29 @@ private void garbageCollectTerminating(Set<Address> addresses, AddressResolver a
         }
     }
 
+    private void checkAndRemoveDrainingBrokers(Set<Address> addresses) throws Exception {
+        BrokerStatusCollector brokerStatusCollector = new BrokerStatusCollector(kubernetes, brokerClientFactory);
+        for (Address address : addresses) {
+            List<BrokerStatus> brokerStatuses = new ArrayList<>();
+            for (BrokerStatus brokerStatus : address.getStatus().getBrokerStatuses()) {
+                if (BrokerState.Draining.equals(brokerStatus.getState())) {
+                    try {
+                        long messageCount = brokerStatusCollector.getQueueMessageCount(address.getAddress(), brokerStatus.getClusterId());
+                        if (messageCount > 0) {
+                            brokerStatuses.add(brokerStatus);
+                        }
+                    } catch (Exception e) {
+                        log.warn("Error checking status of broker {}:{} in state Draining. Keeping.", brokerStatus.getClusterId(), brokerStatus.getContainerId());
```

Comment:
Exception is broad.  Do you think we should log the stack trace so we know exactly why?